src/WindowsPackageManager/ConfigurationStaticFunctions.cpp

```diff
@@ -81,11 +103,78 @@ namespace ConfigurationShim
 
         winrt::Microsoft::Management::Configuration::ConfigurationProcessor CreateConfigurationProcessor(winrt::Microsoft::Management::Configuration::IConfigurationSetProcessorFactory const& factory)
         {
+            if (!m_statics)
+            {
+                THROW_HR(APPINSTALLER_CLI_ERROR_PACKAGE_IS_STUB);
+            }
+
             auto result = m_statics.CreateConfigurationProcessor(factory);
             result.as<AppInstaller::WinRT::ILifetimeWatcher>()->SetLifetimeWatcher(CreateLifetimeWatcher());
             return result;
         }
 
+        bool IsConfigurationEnabled()
+        {
+            return !IsStubPackage();
+        }
+
+        winrt::Windows::Foundation::IAsyncActionWithProgress<uint32_t> EnableConfigurationAsync()
+        {
+            if (IsConfigurationEnabled())
+            {
+                return;
+            }
+
+            co_await winrt::resume_background();
+
+            SetStubPreferred(false);
+
+            PackageManager packageManager;
+            PackageCatalogReference catalogRef{ packageManager.GetPredefinedPackageCatalog(PredefinedPackageCatalog::MicrosoftStore) };
+            THROW_HR_IF(APPINSTALLER_CLI_ERROR_INTERNAL_ERROR, !catalogRef);
+
+            ConnectResult connectResult = catalogRef.Connect();
+            THROW_HR_IF(APPINSTALLER_CLI_ERROR_SOURCE_OPEN_FAILED, connectResult.Status() != ConnectResultStatus::Ok);
+
+            PackageCatalog catalog = connectResult.PackageCatalog();
+
+            FindPackagesOptions findPackagesOptions;
+            PackageMatchFilter filter;
+            filter.Field(PackageMatchField::Id);
+            filter.Option(PackageFieldMatchOption::Equals);
+            filter.Value(AppInstaller::MSStore::s_AppInstallerProductId);
+            findPackagesOptions.Filters().Append(filter);
+
+            FindPackagesResult findPackagesResult{ catalog.FindPackages(findPackagesOptions) };
+
+            auto matches = findPackagesResult.Matches();
+            THROW_HR_IF(APPINSTALLER_CLI_ERROR_MISSING_PACKAGE, matches.Size() == 0);
+            auto catalogPackage = matches.GetAt(0).CatalogPackage();
+
+            InstallOptions installOptions;
+            installOptions.AcceptPackageAgreements(true);
+            installOptions.AllowUpgradeToUnknownVersion(true);
+            installOptions.Force(true);
+
+            auto progress = co_await winrt::get_progress_token();
+
+            // Don't use UpgradePackageAsync
+            auto installTask = packageManager.InstallPackageAsync(catalogPackage, installOptions);
+            installTask.Progress([progress](auto const&, InstallProgress installProgress)
+            {
+                if (installProgress.State == PackageInstallProgressState::Downloading && installProgress.BytesRequired != 0)
+                {
+                    progress((uint32_t)(installProgress.DownloadProgress * 50));

```

Comment:
My guess is that the package download in this case is going to be more like 80% of the time.